module/zfs/zfs_ioctl.c

```diff
@@ -6317,6 +6359,11 @@ zfs_ioctl_init(void)
 	    zfs_ioc_pool_sync, zfs_secpolicy_none, POOL_NAME,
 	    POOL_CHECK_SUSPENDED | POOL_CHECK_READONLY, B_FALSE, B_FALSE);
 
+	zfs_ioctl_register("channel_program", ZFS_IOC_CHANNEL_PROGRAM,
+	    zfs_ioc_channel_program, zfs_secpolicy_config,
```

Comment:
I might be paranoid but perhaps it makes sense to also make sure we have the CAP_SYS_MODULE capacity, instead of just CAP_SYS_ADMIN which is what the zfs_secpolicy_config() checks for.
Might be overkill but reasoning about the security of this becomes easier: if we have CAP_SYS_MODULE we can change the kernel in arbitrary ways anyway so a restricted Lua script is not a problem.

Reply:
Paranoid is good, but I'm not sure `CAP_SYS_MODULE` is appropriate.  Configuring the network stack in Linux also depends on an in-kernel language and they previously used `CAP_SYS_MODULE`  for the reasons you mentioned before switching to `CAP_NET_ADMIN`.  Today `CAP_SYS_MODULE` only covers loading/unloading kmods.  Adding `CAP_NET_ADMIN`  sounds like a good idea, even if the name is misleading.